WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/CoverBlockProcessor.java

```diff
@@ -0,0 +1,78 @@
+package org.wordpress.android.ui.posts.mediauploadcompletionprocessors;
+
+import com.google.gson.JsonObject;
+
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.wordpress.android.util.helpers.MediaFile;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CoverBlockProcessor extends BlockProcessor {
+    /**
+     * Template pattern used to match and splice cover inner blocks
+     */
+    private static final Pattern PATTERN_COVER_INNER = Pattern.compile(new StringBuilder()
+                    .append("(^.*?<div class=\"wp-block-cover__inner-container\">\\s*)")
+                    .append("(.*)") // inner block contents
+                    .append("(\\s*</div>\\s*</div>\\s*<!-- /wp:cover -->.*)").toString(), Pattern.DOTALL);
```

Comment:
oops. I had left a comment here that was wrong. you can take some inspiration from `hpq` which @aduth wrote. something like [jsoup](https://jsoup.org) (random internet search) could automate the inference of the block attributes as defined in the block implementation, rather than falling back to attempts at parsing hTML with RegExp

Reply:
Hi @dmsnell :wave: :smiley: , thanks for taking a look at this one. I'm using jsoup already, which is really helpful. IIRC, I encountered some minor differences in output in certain scenarios, possible because it parses to an AST and not a parse tree, however, it is probably worth revisiting, as the variations might still be acceptable to pass validation. :+1:

Reply:
> I encountered some minor differences in output in certain scenarios, possible because it parses to an AST and not a parse tree

Do you remember what those differences were? Any difference would have to be a bug since we're dealing with HTML/CSS selectors…

Reply:
also it looks like this entire pattern shouldn't be necessary if we process the inner blocks using a spec-compliant parser

Reply:
> Do you remember what those differences were? Any difference would have to be a bug

Off-hand, I recall that jsoup was adding closing tags, which was problematic during at least one of the iterations toward the current implementation. It's likely this is resolved now, as the implementation no longer processes units of non-self-contained HTML, so probably worth revisiting. To be sure, this was not a bug in jsoup, but rather a limitation in a particular iteration of this implementation, in that it required serialization of the parse tree, while jsoup is designed to serialize abstract trees.

> also it looks like this entire pattern shouldn't be necessary if we process the inner blocks using a spec-compliant parser

Indeed, using a spec-compliant parser would obviate the need for this, as we'd be able to handle inner blocks an a much more general way, albeit for _all_ block types.